configure.ac

```diff
@@ -0,0 +1,115 @@
+AC_INIT([mib2h5], [0.0.1], [https://github.com/ePSIC-DLS/mib2h5], [mib2h5],[https://github.com/ePSIC-DLS/mib2h5])
+AM_INIT_AUTOMAKE([-Wall foreign -Werror tar-pax])
+
+# --- Build Type Options --
+AC_ARG_ENABLE([debug],
+  AS_HELP_STRING([--enable-debug], [Enable debug build]),
+  [debug_build=$enableval], [debug_build=yes])
+
+AC_ARG_ENABLE([asan],
+  AS_HELP_STRING([--enable-asan], [Enable AddressSanitizer build]),
+  [asan_build=$enableval], [asan_build=no])
+
+# --- Check for HDF5 ---
+
+AC_ARG_WITH([hdf5],
+  [AS_HELP_STRING([--with-hdf5=PATH], [Path to HDF5 installation])],
+  [HDFDIR="$withval"],
+  [])
+
+AS_IF([test -n "$HDFDIR"], [
+
+  # User gave --with-hdf5 — use it directly
+  CPPFLAGS="$CPPFLAGS -I$HDFDIR/include"
+  LDFLAGS="$LDFLAGS -L$HDFDIR/lib -Wl,-rpath=$HDFDIR/lib"
+
+  AC_CHECK_HEADER([hdf5.h], [], [AC_MSG_ERROR([Missing hdf5.h in $HDFDIR/include])])
+  AC_CHECK_LIB([hdf5], [H5open], [], [AC_MSG_ERROR([Missing libhdf5 in $HDFDIR/lib])])
+  AC_DEFINE([HAVE_HDF5], [1], [Define if HDF5 is available])
+],
+[
+  # No --with-hdf5: try system-wide
+  AC_MSG_CHECKING([for system-wide HDF5 installation])
+  AC_CHECK_HEADER([hdf5.h], [
+    AC_CHECK_LIB([hdf5], [H5open], [
+      AC_DEFINE([HAVE_HDF5], [1], [Define if HDF5 is available])
+    ], [AC_MSG_ERROR([libhdf5 not found system-wide])])
+  ], [AC_MSG_ERROR([hdf5.h not found system-wide])])
+])
+
+# ---------------- Compression Option ----------------
+AC_ARG_WITH([compression],
+  [AS_HELP_STRING([--with-compression], [Enable compression with blosc and hdf5-blosc])],
+  [],
+  [with_compression=check])
```

Comment:
Typically with `AC_ARG_WITH` the option is a path; use `AC_ARG_ENABLE` for a yes/no decision.